### ISSUE TITLE
W2 Tutorial 2 - Collection of issues

### ISSUE THREAD

**Issue by fededagos** [CIS-522/course-content#48]
General feedback and observations from our pod are collected [here](https://docs.google.com/document/d/1hqEygHYy46mNBLtSMg8praunS0bmpKDpQWAh9cb5pkk/edit#).
As a summary: 

- The first section on XOR was hard to follow. Some more explanatory text could be added (virtually after every code box) explaining why we are doing what we are doing (which is only revealed in the end). Especially confusing was the threshold at which non-linearity kicked in.
- Exercises 4-6 all had typos in their final line of code (although the code was correct in the solution)
- The time it took us to complete the tutorial exceeded by far 2.5 hours

**Comment by spirosChv** [comment]
@fededagos I am currently updating the code by taking into account all your comments. Later on, I will update the code on github

**Comment by spirosChv** [comment]
@fededagos Thank you for the comments and finding all of these ugly bugs. I have created a new PR, where I have addresses all of the comments.
